lib/WebSocketRequest.js

```diff
@@ -499,6 +501,11 @@ WebSocketRequest.prototype._handleSocketCloseBeforeAccept = function() {
     this._removeSocketCloseListeners();
 };
 
+WebSocketRequest.prototype._handleSocketErrorBeforeAccept = function(err) {
```

Comment:
why does this only apply to 'BeforeAccept'? are the error handlers overwritten later?

Reply:
Honestly I don't really understand it very well. It looks like this file only accepts or rejects the requests, then it gets handled on another level. So we add the error handler within this scope and remove it once it is accepted or rejected. I assume if an error can happen somewhere outside of this file that might cause an uncaught exception we would have to handle it there